src/VisualStudio/Core/Def/Implementation/ProjectSystem/InvisibleEditor.cs

```diff
@@ -84,6 +84,34 @@ public InvisibleEditor(IServiceProvider serviceProvider, string filePath, IVisua
             }
         }
 
+        private IVsProject TryGetProjectOfHierarchy(IVsHierarchy hierarchyOpt)
+        {
+            // The invisible editor manager will fail in cases where the IVsProject passed to it is not consistent with
+            // the IVsProject known to IVsSolution (e.g. if the object is a wrapper like AbstractHostObject created by
+            // the CPS-based project system). This method returns an IVsProject instance known to the solution, or null
+            // if the project could not be determined.
```

Comment:
Might be worth adding a comment here that null _is_ OK -- it just means we're falling back to the slow path.